sdk/servicebus/azure-servicebus/azure/servicebus/aio/_async_message.py

```diff
@@ -24,15 +19,41 @@
 from .._common.utils import get_running_loop, utc_from_timestamp
 from ..exceptions import MessageSettleFailed
 
+_LOGGER = logging.getLogger(__name__)
+
 
 class ReceivedMessage(sync_message.ReceivedMessage):
     """A Service Bus Message received from service side.
 
     """
 
-    def __init__(self, message, mode=ReceiveSettleMode.PeekLock, loop=None):
+    def __init__(self, message, mode=ReceiveSettleMode.PeekLock, loop=None, **kwargs):
         self._loop = loop or get_running_loop()
-        super(ReceivedMessage, self).__init__(message=message, mode=mode)
+        super(ReceivedMessage, self).__init__(message=message, mode=mode, **kwargs)
+
+    async def _settle_message(
+            self,
+            settle_operation,
+            dead_letter_details=None
+    ):
+        try:
+            if not self._is_deferred_message:
+                try:
+                    await self._loop.run_in_executor(
+                        None,
+                        self._get_settle_via_receiver_link(settle_operation, dead_letter_details)
+                    )
+                    return
+                except Exception as exception:  # pylint: disable=broad-except
+                    _LOGGER.info(
+                        "Message settling: %r has encountered an exception (%r)."
+                        "Trying to settle through management link",
+                        settle_operation,
+                        exception
+                    )
+            await self._get_settle_via_mgmt_link(settle_operation, dead_letter_details)()
```

Comment:
I don't see an async override of this function, does it need to be loop/executor wrapped as well?

Reply:
nope..
the implementation of this method is:
```
return functools.partial(
    self._receiver._settle_message,
    SETTLEMENT_COMPLETE,
    [self.lock_token],
)
```

when async_message call this, `self._receiver._settle_message` is an async method, so functools.partial would also return a async method

Reply:
Ah, clever.